languages/csharp/AppEncryption/AppEncryption/Persistence/AdoMetastorePersistenceImpl.cs

```diff
@@ -23,12 +23,10 @@ public class AdoMetastorePersistenceImpl : IMetastorePersistence<JObject>
 
         private const string LoadQuery = @"SELECT key_record from encryption_key where id = @id and created = @created";
 
-        // Store only if a value with the same key doesn't exist
-        // Using this approach since dotnet doesn't provide a specific integrity violation exception
+        // We try to store even if a value with the same key exists
+        // since dotnet doesn't provide a specific integrity violation exception
         private const string StoreQuery =
-            @"INSERT INTO encryption_key (id, created, key_record) 
-              SELECT @id, @created, @key_record WHERE NOT EXISTS 
-              (SELECT id from encryption_key where id = @id and created = @created )";
+            @"INSERT INTO encryption_key (id, created, key_record) SELECT @id, @created, @key_record";
```

Comment:
since we're just doing a plain insert statement now, we should use the standard `insert into (column1, ...) values (@param1, ...)` style

(the insert select style is typically only used for bulk inserts or where not exists)